src/DataViews/include/DataViews/DataView.h

```diff
@@ -171,6 +171,9 @@ class DataView {
     return nullptr;
   }
 
+  [[nodiscard]] virtual absl::flat_hash_map<std::string_view, bool> GetActionVisibilities(
+      int clicked_index, const std::vector<int>& selected_indices);
+
```

Comment:
This function signature looks like it's breaking the KISS principle - because it's doing an operation on multiple elements at once, but it could also be implemented to operate on a single context menu element. This would also avoid the temporary data structure. It's also not fully clear what the `bool` in the map means. I looked at #3378 and as far as I understand it encode whether this context menu entry will be enabled or not (greyed out), right?

May I suggest an alternative? What about something like this?

```cpp
[[nodiscard]] virtual bool IsActionAvailable(std::string_view action, int clicked_index, const std::vector<int>& selected_indices) const;
[[nodiscard]] virtual bool IsActionVisible(std::string_view action, int clicked_index, const std::vector<int>& selected_indices) const;
```
or

```cpp
enum class ActionStatus { kNotAvailableAndNotVisible, kOnlyVisible, kAvailableAndVisible };

[[nodiscard]] virtual ActionStatus GetActionStatus(std::string_view action, int clicked_index, const std::vector<int>& selected_indices) const;
```

When the context menu is build this function(s) need to be called multiple times - for each possible action. It has the disadvantage that some work has to be done multiple times - like mapping a selected index to a function, but that's not so important because it's just a context menu and affect the user in any way.

Reply:
How about I rename the action status as
```cpp
enum class ActionStatus { kInvisible, kVisibleButDisabled, kVisibleAndEnabled };
```
I used the word `availability` to present `enabled or disabled`. But now we can add more detailed status for the actions so I feel like `Enabled` and `Disabled` will be easier to understand comparing to `Available` and `Unavailable` now.

Reply:
Yes. Great. Way better than what I proposed.